defaults/main.yml

```diff
@@ -23,30 +23,53 @@ datadog_agent_binary_path_windows: "C:\\Program Files\\Datadog\\Datadog Agent\\b
 # list of additional groups for datadog_user
 datadog_additional_groups: {}
 
-# default apt repo
-datadog_apt_repo: "deb https://apt.datadoghq.com/ stable 6"
+# Major version of the Agent that will be installed.
+# Possible values: 5, 6, 7
+datadog_agent_major_version: 7
+
+# Pin agent to a version. Highly recommended.
+# Defaults to the latest version of the major version chosen in datadog_agent_major_version
+# The version pinned needs to have the same major as datadog_agent_major_version
+datadog_agent_version: ""
+
+# default apt repo and keyserver
+
+# By default, will use the official apt Datadog repository for the chosen major version
+# Agent 5: deb https://apt.datadoghq.com/ stable main
+# Agent 6: deb https://apt.datadoghq.com/ stable 6
+# Agent 7: deb https://apt.datadoghq.com/ stable 7
+# Use this option to override the repository used
+datadog_apt_repo: ""
```

Comment:
Having the default in `defaults/main.yml` is better that hardcoding them in the tasks I think. This file is the first place where people look for that kind of information and it avoid the possibility of using an uninitialized var.

Reply:
I'm talking about this https://github.com/DataDog/ansible-datadog/pull/220/files?file-filters%5B%5D=.j2&file-filters%5B%5D=.yml#diff-3cd6ba6341787eb8c9742f9cd62b4c10R46 or this https://github.com/DataDog/ansible-datadog/pull/220/files?file-filters%5B%5D=.j2&file-filters%5B%5D=.yml#diff-3cd6ba6341787eb8c9742f9cd62b4c10R39 where we hardcode default in the task instead of using the defaults/main.yml

Reply:
What I meant is: when would we have an uninitialised var, and when would that be a problem?

Reply:
So I would like to avoid having the repo hard-coded in the task. Also we had issue in the past (related to system-probe for example) where a variable initialized to an empty string in the default cause the role to fail on some edge case after a rework.

We should just move the value from the task to the default and use them. That way we don't have to check id the value is empty in the task.

Reply:
Can you give a use case where the user would have to change the repo URL? I can't see why this is needed.